src/main/scala/org/clulab/asist/agents/DialogAgentReprocessor.scala

```diff
@@ -299,35 +301,43 @@ class DialogAgentReprocessor (
    */
   def reprocessDialogAgentMetadata(s: RunState, line: String): Unit = {
     // parse the metadata text into AST to fix nonstandard JSON issues
-    parseJValue(line).toList.map(metadata => {
-      // if we have a data field, its DialogAgent metadata
-      metadata.findField {
-        case (n: String, data: JObject) => n == "data" 
-        case _ => false
-      }.toList.map(_._2.findField {
-        case (n: String, text: JValue) => n == "text" 
-        case _ => false
-      }).toList.flatten.map(_._2).map (textJValue => {
-        val text: String = textJValue.extract[String]
-        write(  
-          // replace the data.extractions field value
-          metadata.replace( 
-            "data"::"extractions"::Nil,
-            Extraction.decompose(getExtractions(text))
-          )
+    parseJValue(line).toList.map{metadata => 
+      reprocessDialogAgentMetadata(s, line, metadata)
+    }
+  }
+
+  def reprocessDialogAgentMetadata(
+    s: RunState, 
+    line: String,
+    metadata: JValue
+  ): Unit = {
+    // parse the metadata text into AST to fix nonstandard JSON issues
+    // if we have a data field, its DialogAgent metadata
+    metadata.findField {
+      case (n: String, data: JObject) => n == "data" 
```

Comment:
I'm not super familiar with what you're doing here -- but are there really not better ways to look for a given field in a JObject?

Reply:
Yes, the Extractor class (no relation to our work) can do this with key/value mappings.